src/test/ui/type-alias-impl-trait/implied_bounds_closure.stderr

```diff
@@ -0,0 +1,11 @@
+error: lifetime may not live long enough
+  --> $DIR/implied_bounds_closure.rs:17:16
+   |
+LL | fn extend_lifetime<'a>(x: &'a str) -> &'static str {
+   |                    -- lifetime `'a` defined here
+LL |     let t = into_impl(x);
+LL |     helper(|_| t)
+   |                ^ returning this value requires that `'a` must outlive `'static`
```

Comment:
*(Non blocking)* It seems that we are either not properly tracking or not displaying the Obligations involved here, where we should likely be pointing at `helper`'s `'static` obligation introduction.

Reply:
i think this is a consequence of NLL stabilization

Reply:
Interestingly, we don't have *too* much obligation information at all:

```
DEBUG rustc_borrowck::diagnostics::region_errors report_region_error(fr='_#1r, outlived_fr='_#0r)
DEBUG rustc_borrowck::diagnostics::region_errors report_region_error: category=Return(Normal) ObligationCause { span: f43.rs:17:16: 17:17 (#0), body_id: HirId { owner: DefId(0:0 ~ f43[1164]), local_id: 0 }, code: InternedObligationCauseCode { code: None } } None
DEBUG rustc_borrowck::diagnostics::region_errors report_region_error: fr_is_local=true outlived_fr_is_local=false category=Return(Normal)
┐rustc_borrowck::diagnostics::region_errors::report_general_error errci=ErrorConstraintInfo { fr: '_#1r, fr_is_local: true, outlived_fr: '_#0r, outlived_fr_is_local: false, category: Return(Normal), span: f43.rs:17:16: 17:17 (#0) }
┘
```

Reply:
Oh, that's why

https://github.com/rust-lang/rust/blob/3c9765cff18a3c4ba2962ea59d4c3f6966f9700c/compiler/rustc_borrowck/src/region_infer/mod.rs#L2077-L2096